### PR TITLE
Add a lab for Trace Compass Scripting

### PR DESCRIPTION
**Pull request by tahini**
It uses the trace done with custom instrumentation, on which we create
an analysis to save "receiving" and "sending" states and show the
results in a time graph with/without arrows.

Signed-off-by: Geneviève Bastien <gbastien@versatic.net>

### CODE REVIEW

File: labs/204-scripted-analysis-for-custom-instrumentation/scripts/step1_readTrace.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+	// For each event, print the name and the field names
+	eventString = event.getName() + " --> ( "
+
+	var fieldsIterator = event.getContent().getFieldNames().iterator()
+	while (fieldsIterator.hasNext()) {
+		eventString += fieldsIterator.next() + " "
+	}
+	eventString += ")"
+
```
Comment (TheMatthew) [inline labs/204-scripted-analysis-for-custom-instrumentation/scripts/step1_readTrace.js]: how about add a comment here // add code here

File: labs/204-scripted-analysis-for-custom-instrumentation/scripts/step3_stateSystem.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/* The MIT License (MIT)
```
Comment (TheMatthew) [inline labs/204-scripted-analysis-for-custom-instrumentation/scripts/step3_stateSystem.js]: I would put this before "readevent"

File: labs/204-scripted-analysis-for-custom-instrumentation/scripts/step2_readEvents.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+/* The MIT License (MIT)
+ *
+ * Copyright (C) 2019 - Geneviève Bastien <gbastien@versatic.net>
+ */
```
Comment (TheMatthew) [inline labs/204-scripted-analysis-for-custom-instrumentation/scripts/step2_readEvents.js]: Copyright poly too?

File: labs/204-scripted-analysis-for-custom-instrumentation/scripts/step4_timeLine.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
+/* The MIT License (MIT)
```
Comment (TheMatthew) [inline labs/204-scripted-analysis-for-custom-instrumentation/scripts/step4_timeLine.js]: before read-event

Comment (tahini) [reply labs/204-scripted-analysis-for-custom-instrumentation/scripts/step4_timeLine.js]: What do you mean read-event?

**Comment by TheMatthew** [comment]
I'm ok with the answers provided.

**Pull request merged**